solr/core/src/java/org/apache/solr/cloud/api/collections/OverseerCollectionMessageHandler.java

```diff
@@ -113,7 +113,9 @@ public OverseerCollectionMessageHandler(
 
   @Override
   public OverseerSolrResponse processMessage(ZkNodeProps message, String operation) {
-    MDCLoggingContext.setCollection(message.getStr(COLLECTION));
+    // sometimes overseer messages have the collection name in 'name' field, not 'collection'
+    MDCLoggingContext.setCollection(
+        message.getStr(COLLECTION) != null ? message.getStr(COLLECTION) : message.getStr(NAME));
```

Comment:
[small nit] instead of `COLLECTION' which is coming from `CollectionAdminParams` is would be better to use `ZkStateReader.COLLECTION_PROP`.

Reply:
I'll switch.